java/runtime/src/main/java/org/ray/runtime/WorkerContext.java

```diff
@@ -1,133 +1,121 @@
 package org.ray.runtime;
 
 import com.google.common.base.Preconditions;
-import java.util.HashMap;
-import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.concurrent.atomic.AtomicInteger;
 import org.ray.api.id.UniqueId;
 import org.ray.runtime.config.WorkerMode;
 import org.ray.runtime.task.TaskSpec;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 public class WorkerContext {
+
   private static final Logger LOGGER = LoggerFactory.getLogger(WorkerContext.class);
 
-  /**
-   * Worker id.
-   */
   private UniqueId workerId;
 
-  /**
-   * Current task.
-   */
-  private TaskSpec currentTask;
+  private ThreadLocal<UniqueId> currentTaskId;
 
   /**
-   * Current class loader.
+   * Number of objects that have been put from current task.
    */
-  private ClassLoader currentClassLoader;
+  private ThreadLocal<Integer> putIndex;
 
   /**
-   * How many puts have been done by current task.
+   * Number of tasks that have been submitted from current task.
    */
-  private AtomicInteger currentTaskPutCount;
+  private ThreadLocal<Integer> taskIndex;
 
-  /**
-   * How many calls have been done by current task.
-   */
-  private AtomicInteger currentTaskCallCount;
+  private UniqueId currentDriverId;
+
+  private ClassLoader currentClassLoader;
 
   /**
    * The ID of main thread which created the worker context.
    */
   private long mainThreadId;
-  /**
-   * If the multi-threading warning message has been logged.
-   */
-  private AtomicBoolean multiThreadingWarned;
+
 
   public WorkerContext(WorkerMode workerMode, UniqueId driverId) {
-    workerId = workerMode == WorkerMode.DRIVER ? driverId : UniqueId.randomId();
-    currentTaskPutCount = new AtomicInteger(0);
-    currentTaskCallCount = new AtomicInteger(0);
-    currentClassLoader = null;
-    currentTask = createDummyTask(workerMode, driverId);
+    currentTaskId = ThreadLocal.withInitial(UniqueId::randomId);
```

Comment:
Should this be initialized with nil? It looks like it gets overwritten soon after in both branches if the if-block below.

Reply:
I see, thanks!